### DISCUSSION TITLE
Vimeo live stuck in preview instead of showing live video

### DISCUSSION THREAD

**Discussion by douwevdijk** [vimeo/player.js#989] [Bug reports]
### Expected Behavior
Scheduled live events start automatically 
### Actual Behavior
Scheduled vimeo live events stuck on 'not started yet' while the event is already streaming live video ( IS LIVE )
### Steps to Reproduce
_schedule event
_embed stream
_google chrome ( latest )

We are experiencing issues where, embedded, vimeo live events not switching automatically to the live stream once it starts ( it remains stuck in 'event not started yet )'. Users need to refresh the browser first to start the screen. Is anyone else experiencing the same issue? How to resolve this? We cannot have users to refresh the browser.

**Comment by cluster28** [comment]
Same here. The problem is when an AJAX request fails the streaming start checker stops.

version: 2.11.0

**Comment by luwes** [comment]
thanks for the feedback, any chance you could provide a screenshot of the failing requests?

**Comment by anekola** [comment]
![image](https://user-images.githubusercontent.com/1892333/88830576-255a4e00-d19c-11ea-9118-213485da8373.png)

This won't be helpful, but here is how it's stuck for me. We embed this with just the embed code from the Vimeo admin. There are no errors in my console.

**Comment by olatzromeo** [comment]
Good afternoon everyone,

For adding a little more information to fellow Anekola's screenshot...

It seems that when some users access the screen in which the vimeo live event is embedded, after a while (when the transmission has not started yet) the AJAX call made to Vimeo seems to have no answer. In that case the player gets stuck, unable to resume communication and update data with vimeo. In such a way that it can only be resolved by refreshing the page.

It has happened to us with all kinds of devices and browsers, with a good internet connection.

**Comment by craig410** [comment]
We are encountering the same, or similar issue. Posting what I've discovered here incase it helps anyone else.

It appears that when the user opens the page with the embedded video ahead of it's scheduled start, the request made to `https://live-api.cloud.vimeo.com/sessions/` will either respond with
```json
{
    id: "***obscured***"
    provisioner: "studio"
    status: "ended"
    terminate_reason: "provision_expire"
    type: "event"
}
```

or

```json
{
    id: "***obscured***"
    provisioner: "studio"
    status: "active"
    type: "event"
}
```

If the response has status "ended" the client makes no more requests for _1 hour_ (which tallies with the expires timestamp parameter sent in the request)

When it does call the same endpoint again, just inside the 1hr expiry, if it receives the same "ended" response this pattern continues.

Depending on how far out you load the page from the live video starting the third request will be after the initial session expiry, so it calls an endpoint to refresh it and load config which all appear to happen successfully but yet the pattern of 1 hour callbacks continues.

**IF** however you receive the status "active" response, either initially on page load or as a result of the 1 hourly callback, then the client polls the endpoint every 5 seconds so when the video stream goes live, the client picks it up and the video auto-plays as expected. This is why refreshing the page appears to solve the issue.

It's my understanding that the "ended" and "active" responses reflect whether or not the video encoder / streamer we are using is connected to Vimeo at the time the user first lands on the page, whether it is broadcasting live or not is unrelated, it only needs to be connected.

Unfortunately the countdown that is visible to users prior to a scheduled event seems entirely unconnected to the backend logic around polling for video start. This leaves users visiting a page prior to the video going live not any wiser on whether it is going to work or not. In the failure case above, the countdown happily counts to 0 and then displays "this event has not started yet" and will leave the user stuck in this state unless they refresh, or wait for the client's hourly poll to Vimeo. The shorter the duration of the live stream the less likely they will poll in the correct window and see anything at all.

I cannot understand the logic of only polling hourly on a live video with a scheduled start time in the near future, and yet if the encoder is connected the polling is at ~ 5 second intervals.

There is an active support ticket opened with Vimeo for this, can readily reproduce it and I have multiple request archives (HAR) from several browsers I can send if someone from Vimeo wants to reach out for any more info?

tldr;
 - Users need to refresh their browser much closer to the scheduled start time of the video
 - Connect your encoder / device well in advance of going live and keep it alive (as it times-out!)
 - Hope Vimeo decide to poll more frequently than hourly

**Comment by gr8carnac** [comment]
Same issue. Any word on a fix yet?

**Comment by anekola** [comment]
Reported two months ago... this is REALLY disappointing. It's a basic need for a live video platform to actually go live when the event starts. We were basically locked into using Vimeo for this and our 11-day event starts today. Already users are having to refresh their pages AFTER the live video has started. Please prioritize this.

**Comment by luwes** [comment]
Thanks for the feedback, this is really helpful. I'm actively looking into this issue.

**Comment by luwes** [comment]
We have a solution in the pipeline, hopefully it can be released this week.

I just wanted to provide some more info around the issue and offer a workaround that you as the creator can use to avoid this problem. The expiration of the livestream session is causing the player to stop polling for any changes in the state of the livestream. The expiration of the live session is 30 min unless the stream is started, this coincides with the button that appears on the Vimeo site "Are you still there?". The workaround is pretty simple, and that is to start the livestream session less than 30min before actually starting to stream.

A livestream session is created once you press the `Next` button on this screen. Hope this helps.

<img width="1469" alt="Screen Shot 2020-10-01 at 4 47 57 PM" src="https://user-images.githubusercontent.com/360826/94861815-1b580400-0406-11eb-997a-f3d7b392e963.png">

**Comment by luwes** [comment]
hi all, the fix went live today! please let us know if you see any issues on your end. thanks

**Comment by gr8carnac** [comment]
Thank you so much. This was a real problem for my clients. Glad it is fixed

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by douwevdijk** [comment]
> hi all, the fix went live today! please let us know if you see any issues on your end. thanks

@luwes Thank you very much! Can you provide a bit more background info regarding the fix? Does this for example avoid non-polling > 30 min before you go live ( so we do not need a workaround as you mentioned ).

**Comment by luwes** [comment]
yes it should be completely fixed, no need for a workaround. it's fine to activate the live stream session 30min or more before starting to stream. the player will keep polling every 5s, and gets a new live stream session automatically.

**Comment by douwevdijk** [comment]
> yes it should be completely fixed, no need for a workaround. it's fine to activate the live stream session 30min or more before starting to stream. the player will keep polling every 5s, and gets a new live stream session automatically.

Perfect, thanks!

**Comment by CJHCapture** [comment]
Just for everyone's knowledge, this issue is not fixed.  We are streaming from Zoom via RTMP into Vimeo and about 10% of the time, Vimeo gets stuck on "Event has not started".  If you go into the live event setup in Vimeo, the stream is there, but it's not streaming out of vimeo and it still has an active button that says Go Live, which when you click, gets stuck and does nothing.

**Comment by jmccroskoesd** [comment]
I believe this issue is back. Embeds are not refreshing to the live event. Gets stuck on "event has not yet started". Has something broken the auto-polling? @luwes

**Comment by CJHCapture** [comment]
We have given up on this as it is still happening but is too random to track down a reason or pattern.  We spoke to senior engineers at Vimeo and they were unwilling to admit the issue or research any further.  I received no follow-up and/or any further communications from their support, and for this reason, we have abandoned Vimeo and left for another provider.

**Comment by fisherinnovation** [comment]
Thanks for the feedback everyone. I will ticket this for internal review ASAP. Reopening this ticket until it is verified as fixed.

**Comment by bmaculaghHubb** [comment]
@fisherinnovation Just wanted to pop in and say that we have also been experiencing this issue. I was able to capture a console log when it happened the last time.

As others have said, a refresh allowed the stream to play normally but until refreshing it was stuck on "Event has not started"

This was captured while viewing the stream directly from Vimeo itself.


[vimeo.com-1635193893769.log](https://github.com/vimeo/player.js/files/7413423/vimeo.com-1635193893769.log)

**Comment by fisherinnovation** [comment]
@bmaculaghHubb Thanks for sharing!
